sql/set.go

```diff
@@ -80,3 +82,28 @@ func (p *planner) getStringVal(name string, values parser.Exprs) (string, error)
 	}
 	return string(s), nil
 }
+
+func (p *planner) SetTimeZone(n *parser.SetTimeZone) (planNode, error) {
+	switch v := n.Value.(type) {
+	case *parser.CastExpr:
+		d, err := p.evalCtx.EvalExpr(v)
+		if err != nil {
+			return nil, err
+		}
+		if interval, ok := d.(parser.DInterval); ok {
+			p.session.Timezone = &Session_Offset{Offset: int64(interval.Duration)}
+			return &valuesNode{}, nil
+		}
+		return nil, util.Errorf("cannot cast %v to an interval", *v)
+	case parser.DString:
+		location := string(v)
+		if location == "DEFAULT" || location == "LOCAL" {
+			location = "Local"
+		}
+		if _, err := time.LoadLocation(location); err != nil {
```

Comment:
If we're concerned about the performance of loading locations from a string, I expect a time zone of "UTC" would be a common special case which could be set to an offset of zero.